### PR TITLE
Move geom-* specific code to geom-*.R class definition

### PR DESCRIPTION
**Pull request by lazycipher**
So far I've moved the following geom class definitions:
- abline
- point
- text
- ribbon
- bar
- violin, etc

### CODE REVIEW

**Comment by lazycipher** [comment]
Hi @tdhock, I've made some changes, can you review, please?

File: R/geom-contour.r [hunk 1]
```diff
@@ -68,5 +68,10 @@ geom_contour <- function(mapping = NULL, data = NULL,
 #' @include geom-path.r
 GeomContour <- gganimintproto("GeomContour", GeomPath,
   default_aes = aes(weight = 1, colour = "#3366FF", size = 0.5, linetype = 1,
-    alpha = NA)
+    alpha = NA),
+    pre_process = function(g, g.data, ...) {
```
Comment (tdhock) [inline R/geom-contour.r:72]: can you please standardize the indentation? remove spaces before pre_process

Comment (lazycipher) [reply R/geom-contour.r:72]: Just give me a few minutes.

**Comment by tdhock** [comment]
the design looks more reasonable. so you have export_animint in the main Geom, which calls geom-specific pre_process methods?
why is the build failing?
why do you want to merge into r4-fixes? shouldn't you merge into master?

File: R/geom-contour.r (hunk 1 above)
Comment (lazycipher) [reply R/geom-contour.r]: I've updated this.

**Comment by lazycipher** [comment]
Yes right!
The build is failing because of that gist test.
Actually I thought I should be working on this branch and then merge to r-4-fixes and when you'll be back, we'll have just to merge r-4 fixes to master.

Please let me know if you want me to close this PR and create a new one on the master branch.

**Comment by tdhock** [comment]
can you please close this PR, then open a new PR from tdhock:geom-oop to tdhock:master
- creating the branch in tdhock repo should resolve the gist fail issue
- merging to master is what you should do in almost every PR

**Comment by lazycipher** [comment]
I got it! I'll do this!
Thank you so much!

**Pull request closed without merging**